.github/workflows/ci-testing.yml

```diff
@@ -0,0 +1,320 @@
+name: oneDPL CI Testing
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches:
+      - release_oneDPL
+      - main
+      - 'release/**'
+    paths:
+      - '.github/**/*'
+      - 'cmake/**/*'
+      - 'include/**/*'
+      - 'test/**/*'
```

Comment:
This is the important addition, everything else is just separating / moving the jobs as is to a new workflow

Reply:
I guess, we should add `examples `and `make` as well.

Reply:
1) about make:
I don't know reasons why 'make'  would not work.
Make configuration still works, I use it.
But if CI doesn't use make files, we can exclude that folder, of course.

2) "we don't test the examples in the CI."..  
Interesting, it was done intentionally to minimize `CI per commit` time or just we forgot about that?

Reply:
> 1. about make:
>    I don't know reasons why 'make'  would not work.
>    Make configuration still works, I use it.
>    But if CI doesn't use make files, we can exclude that folder, of course.

As far as I understand, the CI generates "fresh" build configuration using `cmake`, then uses that build configuration to build the tests.  Of course, there is nothing wrong with using the make config if it works, it just has not changed much in the last few years and I don't actively use it.  So I'm not aware of its current support for all cases.

> 2. "we don't test the examples in the CI."..
>    Interesting, it was done intentionally to minimize `CI per commit` time or just we forgot about that?

Its possible we build and run the examples and I am unaware of it.  I can try to look into this.  It would be a good idea to do if we aren't already.  I suppose we can add it in here in anticipation of adding these to the CI in the future.

Reply:
Good point about `CMakeLists.txt` ... 
Let me add that.  
I'm OK with leaving out `examples` for now, and adding it if we add it in the future to the CI.